src/ContractInteract/EIP20CoGateway.js

```diff
@@ -0,0 +1,480 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+'use strict';
+
+const Web3 = require('web3');
+const Contracts = require('../Contracts');
+const Utils = require('../../src/utils/Utils');
+const Anchor = require('../../src/ContractInteract/Anchor');
+
+/**
+ * Contract interact for EIP20CoGateway.
+ */
+class EIP20CoGateway {
+  /**
+   * Constructor for EIP20CoGateway.
+   *
+   * @param {Object} web3 Web3 object.
+   * @param {string} coGatewayAddress CoGateway contract address.
+   */
+  constructor(web3, coGatewayAddress) {
+    if (web3 instanceof Web3) {
+      this.web3 = web3;
+    } else {
+      const err = new TypeError(
+        "Mandatory Parameter 'web3' is missing or invalid",
+      );
+      throw err;
+    }
+
+    if (!Web3.utils.isAddress(coGatewayAddress)) {
+      const err = new TypeError(
+        "Mandatory Parameter 'coGatewayAddress' is missing or invalid.",
+      );
+      throw err;
+    }
+
+    this.coGatewayAddress = coGatewayAddress;
+
+    this.contract = Contracts.getEIP20CoGateway(
+      this.web3,
+      this.coGatewayAddress,
+    );
+
+    if (!this.contract) {
+      const err = new TypeError(
```

Comment:
This is probably *not* a `TypeError`, as the type of the address was correct. The value was wrong.
Also in other places/classes.

See also the [nodejs documentation on errors](https://nodejs.org/docs/latest-v9.x/api/errors.html#errors_errors).